it/controllers/benefits/reimbursed/NonTaxableCostsBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,528 @@
+    val expectedIncorrectFormatErrorMessage = "Enter the amount of non-taxable costs reimbursed by your employer in the correct format"
+    val expectedOverMaximumErrorMessage = "The non-taxable costs reimbursed by your employer must be less than £100,000,000,000"
+  }
+
+  object ExpectedIndividualCY extends SpecificExpectedResults {
+    val expectedTitle = "How much of your non-taxable costs were reimbursed by your employer?"
+    val expectedHeading = "How much of your non-taxable costs were reimbursed by your employer?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val expectedNoEntryErrorMessage = "Enter the amount of non-taxable costs reimbursed by your employer"
+    val expectedIncorrectFormatErrorMessage = "Enter the amount of non-taxable costs reimbursed by your employer in the correct format"
+    val expectedOverMaximumErrorMessage = "The non-taxable costs reimbursed by your employer must be less than £100,000,000,000"
+  }
+
+  object ExpectedAgentEN extends SpecificExpectedResults {
+    val expectedTitle = "How much of your client’s non-taxable costs were reimbursed by their employer?"
+    val expectedHeading = "How much of your client’s non-taxable costs were reimbursed by their employer?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val expectedNoEntryErrorMessage = "Enter the amount of non-taxable costs reimbursed by your client’s employer"
+    val expectedIncorrectFormatErrorMessage = "Enter the amount of non-taxable costs reimbursed by your client’s employer in the correct format"
+    val expectedOverMaximumErrorMessage = "The non-taxable costs reimbursed by your client’s employer must be less than £100,000,000,000"
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedTitle = "How much of your client’s non-taxable costs were reimbursed by their employer?"
+    val expectedHeading = "How much of your client’s non-taxable costs were reimbursed by their employer?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val expectedNoEntryErrorMessage = "Enter the amount of non-taxable costs reimbursed by your client’s employer"
+    val expectedIncorrectFormatErrorMessage = "Enter the amount of non-taxable costs reimbursed by your client’s employer in the correct format"
+    val expectedOverMaximumErrorMessage = "The non-taxable costs reimbursed by your client’s employer must be less than £100,000,000,000"
+  }
+
+  private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+  private def employmentUserData(hasPriorBenefits: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+    EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId, isPriorSubmission = true, hasPriorBenefits = hasPriorBenefits, employmentCyaModel)
+
+  def cyaModel(employerName: String, hmrc: Boolean, benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
```

Comment:
for the employerName and hmrc variables, these are always set to the same thing in the tests, so its easier to not pass them into the method, and just set on line 133 employerName to be "employerName" and currentDataIsHmrcHeld to be true.

Reply:
done